src/Utilities/Validator/ExportPreviewValidator.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Omikron\Factfinder\Utilities\Validator;
+
+use Magento\Catalog\Api\ProductRepositoryInterface;
+use Magento\Catalog\Model\Product;
+use Magento\Catalog\Model\Product\Visibility;
+use Magento\ConfigurableProduct\Model\Product\Type\Configurable as ConfigurableType;
+use Omikron\Factfinder\Exception\ExportPreviewValidationException;
+
+class ExportPreviewValidator implements Validator
+{
+    /** @var int */
+    private $entityId;
+
+    /** @var ProductRepositoryInterface */
+    private $productRepository;
+
+    /** @var ConfigurableType */
+    private $configurableType;
+
+    public function __construct(
+        int $entityId,
+        ProductRepositoryInterface $productRepository,
```

Comment:
i don't know if there i a rule for that but i personally sort the arguments that are primitives, to be the last in the whole argument list